apps/demo-pkp-social-auth-next-ts/src/pages/index.tsx

```diff
@@ -0,0 +1,496 @@
+      signMessage: signAuthSig,
+    });
+    setAuthMethod(authMethod);
+    console.log('authMethod', authMethod);
+
+    // Fetch PKPs associated with eth wallet account
+    setView(Views.FETCHING);
+    const pkps: IRelayPKP[] = await litAuthClient.fetchPKPsByAuthMethod(
+      authMethod
+    );
+    if (pkps.length > 0) {
+      console.log(pkps);
+      setPKPs(pkps);
+    }
+    setView(Views.FETCHED);
+  }
+
+  /**
+   * Handle redirect from Lit login server
+   */
+  const handleRedirect = useCallback(async () => {
+    setView(Views.HANDLE_REDIRECT);
+    try {
+      // Get auth method object that has the OAuth token from redirect callback
+      const authMethod: AuthMethod = litAuthClient.handleSignInRedirect();
+      setAuthMethod(authMethod);
+
+      // Fetch PKPs associated with social account
+      setView(Views.FETCHING);
+      const pkps: IRelayPKP[] = await litAuthClient.fetchPKPsByAuthMethod(
+        authMethod
+      );
+      if (pkps.length > 0) {
+        setPKPs(pkps);
+      }
+      setView(Views.FETCHED);
+    } catch (err) {
+      console.error(err);
+      setError(err);
+      setView(Views.ERROR);
+    }
+
+    // Clear url params once we have the OAuth token
+    // Be sure to use the redirect uri route
+    router.replace(window.location.pathname, undefined, { shallow: true });
+  }, [litAuthClient, router]);
+
+  /**
+   * Mint a new PKP for current auth method
+   */
+  async function mint() {
+    setView(Views.MINTING);
+
+    try {
+      // Mint new PKP
+      const newPKP: IRelayPKP = await litAuthClient.mintPKPWithAuthMethod(
+        authMethod
+      );
+
+      // Add new PKP to list of PKPs
+      const morePKPs: IRelayPKP[] = [...pkps, newPKP];
+      setPKPs(morePKPs);
+
+      setView(Views.MINTED);
+      setView(Views.CREATING_SESSION);
+
+      // Get session sigs for new PKP
+      await createSession(newPKP);
+    } catch (err) {
+      console.error(err);
+      setError(err);
+      setView(Views.ERROR);
+    }
+  }
+
+  /**
+   * Generate session sigs for current PKP
+   *
+   * @param {Object} pkp - PKP object
+   */
+  async function createSession(pkp: IRelayPKP) {
+    setView(Views.CREATING_SESSION);
+
+    let authNeededCallback;
+    try {
+      if (authMethod.authMethodType === AuthMethodType.EthWallet) {
+        const signAuthSig = async (message: string) => {
+          const sig = await signMessage({ message: message });
+          return sig;
+        };
+        console.log('litAuthClient.domain', litAuthClient.domain);
+        authNeededCallback = getEthWalletAuthNeededCallback({
+          domain: litAuthClient.domain,
+          address,
+          signMessage: signAuthSig,
+        });
+      } else {
+        authNeededCallback = getSocialAuthNeededCallback({
+          authMethods: [authMethod],
+          pkpPublicKey: pkp.publicKey,
+        });
+      }
+
+      // Get session signatures
+      const sessionSigs = await litNodeClient.getSessionSigs({
```

Comment:
it might be simpler for users to not have to worry about multiple SDKs and only interact with `litAuthClient` - perhaps a light wrapper in `litAuthClient` to wrap around the `litNodeClient.getSessionSigs` call? wdyt?